lib/pure/parseutils.nim

```diff
@@ -569,11 +569,229 @@ proc parseUInt*(s: string, number: var uint, start = 0): int {.
   if result != 0:
     number = uint(res)
 
-proc parseBiggestFloat*(s: string, number: var BiggestFloat, start = 0): int {.
-  magic: "ParseBiggestFloat", importc: "nimParseBiggestFloat", noSideEffect.}
-  ## Parses a float starting at `start` and stores the value into `number`.
-  ## Result is the number of processed chars or 0 if a parsing error
-  ## occurred.
+
+when defined(nimKochBootStrap):
+  proc parseBiggestFloat*(s: string, number: var BiggestFloat, start = 0): int {.
+    magic: "ParseBiggestFloat", importc: "nimParseBiggestFloat", noSideEffect.}
+else:
+  when defined(js):
+    proc parseFloatNative(a: string): float =
+      let a2 = a.cstring
+      asm """
+      `result` = Number(`a2`);
+      """
+
+    #[
+    xxx how come code like this doesn't give IndexDefect ?
+    let z = s[10000] == 'a'
+    ]#
```

Comment:
```suggestion
```
I know you the comment copied from `jssys`, but it belongs neither here nor there.